applicationset/generators/scm_provider.go

```diff
@@ -58,6 +60,21 @@ func (g *SCMProviderGenerator) GetTemplate(appSetGenerator *argoprojiov1alpha1.A
 	return &appSetGenerator.SCMProvider.Template
 }
 
+// Returns an error in case url is not the default provider URL or allowed with allowedScmProviders. If allowedScmProviders is empty it means all are allowed
+func ScmProviderAllowed(url string, allowedScmProviders []string) error {
```

Comment:
Could this return a boolean instead of an error? The error message we're constructing here doesn't really tell the caller anything that a boolean couldn't.